### PR TITLE
[Feature] Deactivate company user

### LINKED ISSUE #187: [Feature] Set inactive process for company users

### PR DESCRIPTION
**Pull request by petersonbrock**
closes #187

### CODE REVIEW

File: api/action-item/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  let {
+        currentOwner, newOwner
+      } = ctx.request.body,
+      update;
+
+  try {
+    update = await ActionItem.update(
+      {
+        owner:       currentOwner,
+        completedOn: null
```
Comment (James1x0) [inline api/action-item/index.js]: With this query, anything passed to the api will get executed. This means that if a user from a company with the "update company users" permission has a session, they can pass a different current/newOwner from a completely different company. To circumvent this security hole, we usually just add a `company: ctx.user.company` to the query

File: api/action-item/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      { owner: newOwner },
+      { multi: true }
+    );
+  } catch (e) {
+    ctx.status = 500;
+    ctx.body = 'Error transferring projects.';
+    return;
+  }
+
+  if (update.nModified === 0) {
```
Comment (James1x0) [inline api/action-item/index.js]: I'd prefer to change this to a 200, with a modified number of 0. Also change L34 to a 200 with modified number.

File: api/action-item/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    update = await ActionItem.update(
+      {
+        owner:       currentOwner,
+        completedOn: null
+      },
+      { owner: newOwner },
+      { multi: true }
+    );
+  } catch (e) {
+    ctx.status = 500;
```
Comment (James1x0) [inline api/action-item/index.js]: This error needs to be logged **as well** via winston so we can manage it in the future. At the top of this file:
```
const { error } = require('winston'); 
```

Somewhere in catch:
```
error(e);
```

File: api/action-item/test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    } = await makeUser({ makeEmployee: true });
+    company = user.company;
+    employee = employeeRecord;
+    testKey = employee._id.toString();
+  });
+
+  after(async () => await mongoose.connection.dropDatabase());
+
+  describe('POST /action-item/bulk-transfer', () => {
+    it('should be secured', expectSecured('post', '/api/v1/action-item/bulk-transfer'));
```
Comment (James1x0) [inline api/action-item/test.js]: Move the function inline here. We're only using it once.

File: api/action-item/index.js (hunk 2 above)
Comment (petersonbrock) [reply api/action-item/index.js]: so there isn't any point of having an if statement for catching zero modified?

**Pull request merged**